docs/adr/0027-document-store-for-payments.md

```diff
@@ -0,0 +1,77 @@
+# Store Payment Data in Firestore
+
+- Deciders: Ben Bangert, Barry Chen
+- Date: 2021-07-22
+
+## Context and Problem Statement
+
+Payment data used in Stripe, PayPal, Apple, and Google IAP are handled as individual documents. We typically need to access different portions of these at different times. In addition, their schema occasionally changes (API updates). Stripe and PayPal have API limits, which means we typically need to duplicate/cache data we retrieve from them. At the moment we cache this data in Redis. This mostly works, but restricts our ability to run queries across payment data sets without pulling it all into BigQuery, or making substantial amounts of Stripe/PayPal requests.
+
+To handle IAP, we need to store Apple receipts and Google subscription objects. These should be organized with the FxA user they belong to. They're arbitrary JSON documents that we also need copied to BigQuery for ETL jobs. To future-proof our payment data storage needs we should have the flexibility needed to store Stripe/Paypal payment data in the same data store with the same ETL capability.
+
+## Decision Drivers
+
+- Engineering resources
+- Arbitrary JSON document querying ability
+- Ease of loading data for ETL jobs
+- Capable of storing arbitrary JSON data
+- Integration with Stripe and other payment systems
+
+## Considered Options
+
+- Google Firestore
+- MySQL
+- Redis
```

Comment:
In our 1:1 yesterday, you noted one additional option we could consider: MongoDB.

Also, I was reading this [Decision Brief: Technology Recommendations for Web Services](https://docs.google.com/document/d/1wGB7qbARy_ie-9YURLii2gDT7gy6kR6nsMhXAzo0VnM/edit?usp=sharing), and it noted that GCP has a few databases: Cloud Spanner, BigTable, and Firestore. Did you also consider Cloud Spanner and BigTable?